cdci_data_analysis/analysis/email_helper.py

```diff
@@ -353,25 +353,53 @@ def store_email_api_code_attachment(api_code, status, scratch_dir, sending_time=
     return attachment_file_path
 
 
-def is_email_to_send_run_query(logger, status, time_original_request, scratch_dir, job_id, config, decoded_token=None):
+def log_email_sending_info(logger, status, time_request, scratch_dir, job_id, additional_info_obj=None):
+    path_email_history_folder = os.path.join(scratch_dir, 'email_history')
+    if not os.path.exists(path_email_history_folder):
+        os.makedirs(path_email_history_folder)
+    history_info_obj = dict(time=timestamp2isot(time_request),
+                            status=status,
+                            job_id=job_id)
+    if additional_info_obj is not None:
+        history_info_obj['additional_information'] = additional_info_obj
+    email_history_log_fn = os.path.join(path_email_history_folder, f'email_history_log_{time_request}.log')
```

Comment:
this filename is not sufficiently unique. Please add some other fields: e.g. status, and also maybe just add a random hash.

Reply:
it includes now status and the hash of the record object